api/monitor/monitorclient/handler.go

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package monitorclient
+
+import (
+	"context"
+
+	"github.com/TheThingsNetwork/go-utils/grpc/sendbuffer"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// HandlerClient creates a new client for Handler monitoring
+func (m *MonitorClient) HandlerClient(ctx context.Context, opts ...grpc.CallOption) Stream {
+	c := new(componentClient)
+	c.log = m.log.WithField("Component", "Handler")
+	if md, ok := metadata.FromOutgoingContext(ctx); ok {
+		if id, ok := md["id"]; ok && len(id) > 0 {
+			c.log = c.log.WithField("ID", id[0])
+		}
+	}
+	c.setup = func() {
+		ctx, c.cancel = context.WithCancel(ctx)
+		for name, cli := range m.clients {
+			name, cli := name, cli // shadow vars
+
+			uplink := sendbuffer.New(m.bufferSize, func() (grpc.ClientStream, error) {
```

Comment:
names like this make it hard to undestand, what's actually going on IMO. why not call it `stream`,  `buf`/`buffer`/`b`, something in that direction? I have no clue what `sendbuffer` does, so I leave that up to you.